corehq/apps/userreports/reports/data_source.py

```diff
@@ -157,8 +158,16 @@ def has_total_row(self):
         return any(column_config.calculate_total for column_config in self.column_configs)
 
     def get_total_records(self):
-        # TODO - actually use sqlagg to get a count of rows
-        return len(self.get_data())
+        qc = self.query_context()
+        for c in self.columns:
+            qc.append_column(c.view)
+
+        session = connection_manager.get_scoped_session(self.engine_id)
+        try:
+            return qc.count(session.connection(), self.filter_values)
+        except:
+            session.rollback()
```

Comment:
interesting. it looks like the call shouldn't do anything if there's no open transaction so probably fine to leave (although could be better to remove and only add if it causes a problem of some kind)